pyramid_swagger/ingest.py

```diff
@@ -62,27 +82,76 @@ def _load_resource_listing(resource_listing):
         )
 
 
-def compile_swagger_schema(schema_dir):
+def generate_resource_listing(schema_dir, listing_base):
+    if 'apis' in listing_base:
+        raise ResourceListingGenerationError(
+            "{0}/{1} has an `apis` listing. Generating a listing would "
+            "override this listing.".format(schema_dir, API_DOCS_FILENAME))
+
+    return dict(
+        listing_base,
+        apis=[{'path': path} for path in find_resource_paths(schema_dir)]
+    )
+
+
+def get_resource_listing(schema_dir, should_generate_resource_listing):
+    """Return the resource listing document.
+
+    :param schema_dir: the directory which contains swagger spec files
+    :type  schema_dir: string
+    :param should_generate_resource_listing: when True a resource listing will
+        be generated from the list of *.json files in the schema_dir. Otherwise
+        return the contents of the resource listing file
+    :type should_enerate_resource_listing: boolean
+    :returns: the contents of a resource listing document
+    """
+    listing_filename = os.path.join(schema_dir, API_DOCS_FILENAME)
+    resource_listing = _load_resource_listing(listing_filename)
```

Comment:
Even with the auto-gen feature, you still need a "base" resource listing file, that includes all the other keys (everything except for `apis`. Which is at least the `swaggerVersion`, but could also include a few other keys (apiVersion, authentication, etc).

I wasn't sure how to properly express that. Maybe this needs another section in the docs to fully explain how it works (with some examples) ?

Maybe this should be renamed to `generate_api_listing` instead of `generate_resource_listing` ?